### PR TITLE
Koji builds

### PR DESCRIPTION
**Pull request by joejstuart**

### CODE REVIEW

File: JenkinsfileBuildTrigger [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                                  remote: repo[1]])
+
+}
+
+properties(
+        [
+                buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '500', daysToKeepStr: '', numToKeepStr: '500')),
+                parameters(
+                        [
+                                string(description: 'Red Hat UMB Msg', defaultValue: '{}', name: 'CI_MESSAGE')
```
Comment (johnbieren) [inline JenkinsfileBuildTrigger]: This is upstream so no Red Hat UMB stuff. This is the fedmsg

File: JenkinsfileBuildTrigger [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                stepName = 'schedule build'
+                buildJob = "fedora-${targetRelease}-build-pipeline"
+                stage(stepName) {
+
+                    retry(TRIGGER_RETRY_COUNT) {
+                        packagepipelineUtils.timedPipelineStep(stepName: stepName, debug: true) {
+
+                            build job: buildJob,
+                                    parameters: [string(name: 'PROVIDED_KOJI_TASKID', value: env.fed_msg_task_id),
+                                                 string(name: 'ARTIFACT_TYPE', value: buildJob)],
```
Comment (johnbieren) [inline JenkinsfileBuildTrigger]: We have no storage behind the all package pipelines so there is no ARTIFACT_TYPE. Everything is just in workspace

File: src/org/centos/pipeline/PackagePipelineUtils.groovy [hunk 3]
```diff
@@ -306,21 +321,37 @@ def checkBranch() {
...
+/**
+ * Parse the repo name from the request list.
+ * Sets env.fed_repo
+ * @param request
+ * @return
+ */
+def repoFromRequest(def request) {
+    if (!env.fed_repo) {
+        try {
+            def pkgUrlTok = request[0].tokenize('/')
```
Comment (johnbieren) [inline src/org/centos/pipeline/PackagePipelineUtils.groovy]: I am looking a little more at the message example you send me. I think we can just use name. It seems name, version, and release are all in the message, and name is the repo name *I think*. There is some discrepancy like for krb5 that I am not entirely clear on, so I think we should check with Miroslav on if we can use rpm name as the repo or we need to do this

File: src/org/centos/pipeline/PackagePipelineUtils.groovy [hunk 4]
```diff
@@ -306,21 +321,37 @@ def checkBranch() {
...
+def checkRelease(String msgRelease) {
+    def targetRelease = null
 
-    return [measurement, tags, fields]
-}
+    def release = msgRelease.tokenize('.').last()
+    if (release ==~ /fc[2-9][0-9]/) {
+        targetRelease = release[2, 3]
+    } else {
+        println "Invalid release"
```
Comment (johnbieren) [inline src/org/centos/pipeline/PackagePipelineUtils.groovy]: I think something more along the lines of we don't care about this release would be a little better. Would say it doesn't matter, but if we are storing 500 builds for debugging purposes, might as well be a little more explicit

File: src/org/centos/pipeline/PackagePipelineUtils.groovy [hunk 5]
```diff
@@ -306,21 +322,40 @@ def checkBranch() {
...
+ */
+def checkRelease() {
+    def targetRelease = null
 
-    return [measurement, tags, fields]
-}
+    if (env.fed_msg_release) {
+        def release = env.fed_msg_release.tokenize('.').last()
+        if (release ==~ /fc[2-9][0-9]/) {
+            targetRelease = release[2, 3]
```
Comment (johnbieren) [inline src/org/centos/pipeline/PackagePipelineUtils.groovy]: I think targetRelease = "f" + release[2, 3]

File: JenkinsfileBuildTrigger [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            stepName = 'upstream-fedora-pipeline-build-trigger'
+            stage(stepName) {
+
+                packagepipelineUtils.timedPipelineStep(stepName: stepName, debug: true) {
+
+                    print "CI_MESSAGE"
+                    print CI_MESSAGE
+                    pipelineUtils.flattenJSON('fed', env.CI_MESSAGE)
+                    packagepipelineUtils.repoFromRequest()
+                    pipelineUtils.updateBuildDisplayAndDescription()
```
Comment (johnbieren) [inline JenkinsfileBuildTrigger]: updateBuildDisplayandDescription is the same thing as setBuildDisplayandDescription without the else. So, I think setBuildDisplayAndDescription is better in all cases. I have actually removed the update one from our other work. Don't have to change it, just my thoughts

File: Jenkinsfile [hunk 7]
```diff
@@ -210,7 +214,7 @@ podTemplate(name: podName,
                             packagepipelineUtils.setStageEnvVars(currentStage)
 
                             // Return a map (messageFields) of our message topic, properties, and content
-                            messageFields = packagepipelineUtils.setMessageFields("package.running")
+                            messageFields = packagepipelineUtils.setMessageFields("package.running", artifact)
 
                             // Send message org.centos.prod.ci.pipeline.allpackages.package.running on fedmsg
```
Comment (johnbieren) [inline Jenkinsfile]: tedious but updating the comments to reflect new field too would be cool.

We will need to update the docs quite a bit soon anyways though so we can just do that with docs update doesn't matter to me

File: src/org/centos/pipeline/PackagePipelineUtils.groovy (hunk 5 above)
Comment (joejstuart) [reply src/org/centos/pipeline/PackagePipelineUtils.groovy]: yep. good catch.

**Pull request merged**